omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,144 @@
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_0.ALLERGY_ID; // allergy does not support uuid
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return 4;
+	}
+	
+	@Test
+	public void deleteUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergy(AALLERGY_ID);
+		Assert.assertFalse(allergy.isVoided());
+		
+		handle(newDeleteRequest(getURI() + "/" + RestTestConstants2_0.ALLERGY_ID, 
+				new Parameter("reason", "unit test")));
+		allergy = Context.getPatientService().getAllergy(AALLERGY_ID);
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+
+	@Test
+	//@Ignore("REST API does not support delete /patient/patient_uuid/allergy/ to delete all allergies")
+	public void deleteAllAllergies() throws Exception
+	{
+		handle(newDeleteRequest(getURI(), 
+				new Parameter("reason", "unit test")));
+		Allergy allergy = Context.getPatientService().getAllergy(AALLERGY_ID);
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	/*
+	@Test
+	@Ignore("REST API does not support PURGE")
+	public void purgeUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergy(AALLERGY_ID);
+		Assert.assertFalse(allergy.isVoided());
+		
+		handle(newDeleteRequest(getURI() + "/" + RestTestConstants2_0.ALLERGY_ID, new Parameter("purge", "")));
+		Assert.assertNull(Context.getPatientService().getAllergy(AALLERGY_ID));
+	}
+	
+	@Test
+	@Ignore("REST API does not support PURGE")
+	public void purgeAllAllergies() throws Exception
+	{
+		handle(newDeleteRequest(getURI() + "/", new Parameter("purge", "")));
+		Assert.assertNull(Context.getPatientService().getAllergy(AALLERGY_ID));
+	}
+	*/
+	
+	@Test
+	public void saveAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergy(AALLERGY_ID);
+		Patient patient = allergy.getPatient();
+
+		DefaultRepresentation rep = new DefaultRepresentation();		
+
+		SimpleObject allergyObject =  new SimpleObject();
+		allergyObject.add("allergyId", 10);
+		allergyObject.add("comment", allergy.getComment());
+
+		Concept codedAllergenConcept = Context.getConceptService().getConceptByUuid("35d3346a-6769-4d52-823f-b4b234bac3e3");
+
+		SimpleObject allergenObject = new SimpleObject();
+		allergenObject.add("allergenType", AllergenType.ANIMAL);
+		allergenObject.add("nonCodedAllergen", "test allergen");
+		ConceptResource1_11 conceptResource = (ConceptResource1_11) Context.getService(RestService.class).getResourceBySupportedClass(Concept.class);
+		allergenObject.add("codedAllergen", conceptResource.asRepresentation(codedAllergenConcept, rep));
```

Comment:
This should actually fail because you are setting both a nonCodedAllergen and a codedAllergen. If it succeeds that's a bug in the underlying API.